### PR TITLE
Refactor for v3

### LINKED ISSUE #12: Incompatible with Companion 3.0?
Seems this module isn't included in Companion 3.0 as default plug-in and it stopped working with 

`throw new Error("addUpgradeScript has been removed and replaced by a new static GetUpgradeScripts flow. Check the wiki for more information")`

when you try to use it. Any plans to make it work with 3.0 or is this module now unmaintained?

Jan

### PR DESCRIPTION
**Pull request by MegaAdragon**
Refactor the module for companion v3

-> try to resolve #12

### CODE REVIEW

File: companion/manifest.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+	"id": "phillips-hue",
```
Comment (MegaAdragon) [inline companion/manifest.json:2]: Keep this type in order to overwrite the old module

File: index.js [hunk 2]
```diff
@@ -1,497 +1,191 @@
...
+		}
+
+		this.updateActions() // export actions
+	}
+	// When module gets deleted
+	async destroy() {
+		this.log('debug', 'destroy')
+	}
+
+	async configUpdated(config) {
```
Comment (MegaAdragon) [inline index.js:57]: Somehow this is never called... Not sure why...

**Comment by MegaAdragon** [comment]
@josephdadams do you want to review the changes?
Or can I just merge the branch with all the changes?

I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…

File: companion/manifest.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"maintainers": [
+		{
+			"name": "Dominik Zipperle",
+			"email": "dominik.zipperle@googlemail.com"
+		}
+	],
+	"runtime": {
+		"type": "node18",
+		"api": "nodejs-ipc",
+		"apiVersion": "1.0.0",
```
Comment (Julusian) [inline companion/manifest.json:19]:
```suggestion
		"apiVersion": "0.0.0",
```

This is why `configUpdated` is never called

File: package.json [hunk 4]
```diff
@@ -1,30 +1,21 @@
...
-	"keywords": [
-		"Philips",
-		"philips-hue"
-	],
-	"manufacturer": "Philips",
-	"product": "Philips Hue",
-	"shortname": "philips-hue",
-	"description": "Philips Hue module for Companion",
+	"name": "philips-hue",
+	"version": "0.2.0",
```
Comment (Julusian) [inline package.json:3]: this should probably be bumped to 1.0.0 now

File: companion/manifest.json (hunk 1 above)
Comment (Julusian) [reply companion/manifest.json:2]: you can change this if you add the old spelling to the `legacyIds` field further down

**Comment by Julusian** [comment]
> I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…

Yeah, it looks like every action has been renamed, so existing users will have all their actions break when they update companion to the version that includes this.

It would be a good idea to write an upgradescript to translate existing configs across. Something like https://github.com/bitfocus/companion-module-bmd-atem/blob/master/src/upgrades.ts#L131-L149, but perhaps with some updating of the options too

**Comment by gauthierGrsse** [comment]
> Refactor the module for companion v3
> 
> -> try to resolve #12

Do you have some news for this bug fix for V3.0 ?

**Comment by josephdadams** [comment]
I believe we are waiting on the PR author to add the upgrade scripts so that this doesn’t break the config for the hundreds of users

**Comment by MegaAdragon** [comment]
> > I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…
> 
> Yeah, it looks like every action has been renamed, so existing users will have all their actions break when they update companion to the version that includes this. For reference, there are at least a few hundred users of this module
> 
> It would be a good idea to write an upgradescript to translate existing configs across. Something like https://github.com/bitfocus/companion-module-bmd-atem/blob/master/src/upgrades.ts#L131-L149, but perhaps with some updating of the options too

@Julusian 
I looked at https://github.com/bitfocus/companion-module-base/wiki/Upgrade-scripts and tried to figure out an upgrade script, but I did not really come far... I found no possibility the debug the script properly.

Because I refactored + cleaned up the actions  completely, I would propose to just accept the broken actions for users that come from the old version of this module. I'm working on feedbacks and presets, so there is in actual benefit in manually restoring the button.

What do you think?

**Comment by MegaAdragon** [comment]
Resolve #10 
and 
Resolve #16 (duplicate for #12)

**Comment by MegaAdragon** [comment]
From my side, I'm ready to merge this branch. As mentioned, this will break configs from previous versions, but I completely overhauled the module...

**Comment by SirAuran** [comment]
Hi are there any updates? Thanks a lot for the effort.

**Pull request merged**